### PR TITLE
finishes diamond function with all tests passing

### PR DESCRIPTION
**Pull request by mistakeets**

### CODE REVIEW

File: exercism/diamond/diamond.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+var Diamond = function() {}
```
Comment (FrankieFabuloso) [inline exercism/diamond/diamond.js]: Don't use var. 
let and const offer more than var does (scoping, clear distinction between immutability and mutability, etc).
Try to use ES6 where you can, just to practice.

File: exercism/diamond/diamond.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+Diamond.prototype.makeDiamond = function(letter) {
+  if (letter === 'A') {
+    return 'A\n'
+  }
+  var alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ".split('')
+  var character = ' '
+  var middle = alphabet.indexOf(letter) + 1
```
Comment (FrankieFabuloso) [inline exercism/diamond/diamond.js]: middleOfDiamond would be a better name for this.
Try and make your variables as meaningful as possible.
It took me a while to figure out what you meant by middle.

File: exercism/diamond/diamond.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+var Diamond = function() {}
+
+Diamond.prototype.makeDiamond = function(letter) {
+  if (letter === 'A') {
+    return 'A\n'
+  }
+  var alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ".split('')
+  var character = ' '
```
Comment (FrankieFabuloso) [inline exercism/diamond/diamond.js]: The variable 'character' is unnecessary.
This, I believe, is an instance where explicity passing in `' '` below during your join is better than passing in a variable.

Either than or don't name it "character", name it something like "whitespace"

File: exercism/diamond/diamond.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+  for (var i = 1; i < middle; i++) {
+    tempStorage[i].push(alphabet[i])
+  }
+
+  for (var i = 0; i < middle; i++) {
+    var spacesBeforeAndAfter = new Array((middle) - i).join(character)
+    tempStorage[i].push(alphabet[i])
+    tempStorage[i].unshift(spacesBeforeAndAfter)
+    tempStorage[i].push(spacesBeforeAndAfter)
```
Comment (FrankieFabuloso) [inline exercism/diamond/diamond.js]: This is pretty cool! I would have never though to just push on either side! Nice 👍

File: exercism/diamond/diamond.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  }
+
+  tempStorage[1].splice(2, 0, character)
+  for (var x = 2; x < middle; x++) {
+    var spacesBetweenLetters = new Array(3 + (x * 2) - 3).join(character)
```
Comment (FrankieFabuloso) [inline exercism/diamond/diamond.js]: I would write the math in a small helper function that is named on what is actually happening. Cause I have no idea what `new Array(3 + (x * 2) - 3).join(character)` is actually doing.

File: exercism/diamond/diamond.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    tempStorage[x].splice(2, 0, spacesBetweenLetters)
+  }
+
+  var result = []
+  var assembleLine = ''
+  for (i = 0; i < middle; i++) {
+    assembleLine = tempStorage[i].join('')
+    result.push(assembleLine)
+  }
+  for (k = result.length - 2; k >= 0; k--) {
```
Comment (FrankieFabuloso) [inline exercism/diamond/diamond.js]: why not 'k' all of a sudden as your iterator?

**Review by FrankieFabuloso — COMMENTED** [review]
Hey left some comments. 😄

File: exercism/diamond/diamond.js (hunk 4 above)
Comment (mistakeets) [reply exercism/diamond/diamond.js]: thanks!

**Pull request merged**